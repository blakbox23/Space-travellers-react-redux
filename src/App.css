* {
  margin: 0;
  padding: 0;
}

.flex {
  display: flex;
}

.header-section {
  border-bottom: 1px solid black;
  height: 50px;
  justify-content: space-between;
  align-items: center;
}

.nav-links {
  list-style-type: none;
}

.nav-links li {
  margin-left: 1.5rem;
}