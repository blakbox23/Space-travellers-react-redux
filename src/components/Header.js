import React from 'react';
import { NavLink } from 'react-router-dom';

function Header() {
  return (
    <div className="header-section flex">
      <div className="logo">
        <img src="https://cdn-icons-png.flaticon.com/512/3212/3212567.png" alt="logo-img" className="logo-img" />
        <span> Space Traveler&apos;s Hub </span>
      </div>
      <ul className="nav-links flex">
        <li>
          <NavLink to="/">Rockets</NavLink>
        </li>
        <li>
          <NavLink to="/missions">Missions</NavLink>
        </li>
        <li>
          <NavLink to="/profile">My Profile</NavLink>
        </li>

      </ul>
    </div>
  );
}

export default Header;
