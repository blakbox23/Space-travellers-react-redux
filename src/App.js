import './App.css';
import { Route, Switch } from 'react-router-dom';
import { Provider } from 'react-redux';
import Header from './components/Header';
import Missions from './components/Missions';
import Profile from './components/Profile';
import Rockets from './components/Rockets';
import store from './redux/configureStore';

function App() {
  return (
    <div>
      <Provider store={store}>
        <Header />
        <Switch>
          <Route exact path="/">
            <Rockets />
          </Route>
          <Route exact path="/missions">
            <Missions />
          </Route>
          <Route exact path="/profile">
            <Profile />
          </Route>
        </Switch>
      </Provider>
    </div>
  );
}

export default App;
