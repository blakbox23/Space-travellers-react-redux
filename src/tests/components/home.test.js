import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom/extend-expect';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from '../../components/Header';
import App from '../../App';

it('should render the header', () => {
  const component = render(<Router><Header /></Router>);
  expect(component).toMatchSnapshot();
});

describe('Click events', () => {
  test('expect rockets to be default page when the app starts', () => {
    render(<Router><App /></Router>);
    expect(screen.queryByText(/All Rockets/)).toBeInTheDocument();
  });

  test('that user to redirected to the Mission page on clicking Quote', () => {
    render(<Router><App /></Router>);
    expect(screen.queryByText(/Status/)).toBeNull();
    userEvent.click(screen.getByText('Missions'));
    expect(screen.queryByText(/Status/)).toBeInTheDocument();
  });

  test('that user to redirected to the PROFILE page on clicking profile', () => {
    render(<Router><App /></Router>);
    expect(screen.queryByText(/My Rockets/)).toBeNull();
    userEvent.click(screen.getByText('My Profile'));
    expect(screen.queryByText(/My Rockets/)).toBeInTheDocument();
  });
});
