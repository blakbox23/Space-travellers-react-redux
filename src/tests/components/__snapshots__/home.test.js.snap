// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the header 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="header-section flex"
      >
        <div
          class="logo"
        >
          <img
            alt="logo-img"
            class="logo-img"
            src="https://cdn-icons-png.flaticon.com/512/3212/3212567.png"
          />
          <span
            class="fs-3"
          >
             Space Traveler's Hub 
          </span>
        </div>
        <ul
          class="nav-links flex"
        >
          <li>
            <a
              aria-current="page"
              class="active-link"
              href="/"
            >
              Rockets
            </a>
          </li>
          <li>
            <a
              href="/missions"
            >
              Missions
            </a>
          </li>
          <li>
            <a
              href="/profile"
            >
              My Profile
            </a>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="header-section flex"
    >
      <div
        class="logo"
      >
        <img
          alt="logo-img"
          class="logo-img"
          src="https://cdn-icons-png.flaticon.com/512/3212/3212567.png"
        />
        <span
          class="fs-3"
        >
           Space Traveler's Hub 
        </span>
      </div>
      <ul
        class="nav-links flex"
      >
        <li>
          <a
            aria-current="page"
            class="active-link"
            href="/"
          >
            Rockets
          </a>
        </li>
        <li>
          <a
            href="/missions"
          >
            Missions
          </a>
        </li>
        <li>
          <a
            href="/profile"
          >
            My Profile
          </a>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
